src/sdk/namespace_fs.js

```diff
@@ -71,6 +74,63 @@ function make_named_dirent(name) {
     return entry;
 }
 
+// function check_md_conditions(conditions, stat) {
+//     if (!conditions) return;
+//     if (!conditions.if_match_etag &&
+//         !conditions.if_none_match_etag &&
+//         !conditions.if_modified_since &&
+//         !conditions.if_unmodified_since) return;
+
+//     const data = stat ? {
+//         etag: this._get_etag(stat),
+//         last_modified: stat.mtime
+//     } : {
+//         etag: '',
+//         last_modified: 0,
+//     };
+
+//     // See http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectHEAD.html#req-header-consideration-1
+//     // See https://tools.ietf.org/html/rfc7232 (HTTP Conditional Requests)
+//     let matched = false;
+//     let unmatched = false;
+
+//     if (conditions.if_match_etag) {
+//         if (!(stat && http_utils.match_etag(conditions.if_match_etag, data.etag))) {
+//             throw Object.assign(new Error('check_md_conditions failed'), { rpc_code: 'IF_MATCH_ETAG', rpc_data: data });
+//         }
+//         matched = true;
+//     }
+//     if (conditions.if_none_match_etag) {
+//         if (stat && http_utils.match_etag(conditions.if_none_match_etag, data.etag)) {
+//             throw Object.assign(new Error('check_md_conditions failed'), { rpc_code: 'IF_NONE_MATCH_ETAG', rpc_data: data });
+//         }
+//         unmatched = true;
+//     }
+//     if (conditions.if_modified_since) {
+//         if (!unmatched && (!stat || conditions.if_modified_since > data.last_modified)) {
+//             throw Object.assign(new Error('check_md_conditions failed'), { rpc_code: 'IF_MODIFIED_SINCE', rpc_data: data });
+//         }
+//     }
+//     if (conditions.if_unmodified_since) {
+//         if (!matched && (!stat || conditions.if_unmodified_since < data.last_modified)) {
+//             throw Object.assign(new Error('check_md_conditions failed'), { rpc_code: 'IF_UNMODIFIED_SINCE', rpc_data: data });
+//         }
+//     }
+// }
+
+function get_xattr_read(xattr) {
+    return _.transform(xattr, (result, value, key) => {
+        const key_str = String(key);
+        if (key_str.startsWith(XATTR_USER_PREFIX)) result[key_str.slice(XATTR_USER_PREFIX.length)] = value;
+    }, {});
+}
+
+function get_xattr_write(xattr) {
```

Comment:
Rename and add the empty check inside:
```suggestion
function get_fs_xattr_for_write(xattr) {
    if (_.isEmpty(xattr)) return undefined;
```